driver-core/src/main/java/com/datastax/driver/core/querybuilder/Select.java

```diff
@@ -95,7 +95,7 @@ StringBuilder buildQueryString(List<Object> variables, CodecRegistry codecRegist
             Utils.joinAndAppend(builder, codecRegistry, " AND ", where.clauses, variables);
         }
 
-        if (orderings != null) {
+        if (orderings != null && !orderings.isEmpty()) {
```

Comment:
I don't think we should silently ignore if the user calls QueryBuilder requests that are not valid to create a CQL query. If the argument in the `orderBy` method is not correct for a CQL query we should throw an exception. As it is done for other query builder methods, like `limit()` if the argument is not valid for Cassandra (limit < 0), we should not ignore it in the query

Reply:
You've certainly fixed the issue as reported in JIRA.  My only feedback would be that when the person writing code against the querybuilder doesn't know exactly what the end user is going to give them as a query, not accepting and empty list of ordering can make for some ugly builder code:

https://gist.github.com/podnov/b67f7e57cf702df579b68b67dde839c7